### ISSUE TITLE
Daemon made invalid request

### ISSUE THREAD

**Issue by ephopper** [redgeoff/spiegel#148]
I'm receiving the below errors for the listener and replicator processes. 
the replication still seems to work. It appears that the errors for those ([line 195](https://github.com/redgeoff/spiegel/blob/0768746c31f56bda65c98f288cd4ae1dfbbd2e92/src/replicators.js#L195)
and [line 199](https://github.com/redgeoff/spiegel/blob/0768746c31f56bda65c98f288cd4ae1dfbbd2e92/src/replicators.js#L199)) refer to log output.

the update listener also seems to be keeping track of the last sequence number and the error also refers to log output ([line 206](https://github.com/redgeoff/spiegel/blob/0768746c31f56bda65c98f288cd4ae1dfbbd2e92/src/update-listeners.js#L206))

the change listener ~~though does not function~~ seems to be functioning as well, but still logs an error. The error refers to [this](https://github.com/redgeoff/spiegel/blob/0768746c31f56bda65c98f288cd4ae1dfbbd2e92/src/change-processor.js#L107)

```
[error] 2018-07-28T00:18:44.473190Z couchdb@172.31.16.234 <0.218.0> -------- Daemon "spiegelUpdates" made invalid request: {[{<<"name">>,<<"spiegel">>},{<<"hostname">>,<<"localhost">>},{<<"pid">>,10802},{<<"level">>,30},{<<"msg">>,<<"Saving lastSeq= 1443-g1AAAAKmeJyl0EsOgjAQBuDKI7Jw61pPQCiFtqzkJtpnCEFdudZ4Eb2J3kRvgn2QuDImsJlJZvJ_mUwHAEiaUIKlOJ5EI3kNSZ4imEKa5hB3Zh0wwFd937dNyOK9GcyLQmoM1a_QH46vTeWbQQycWGHNMoHHirUVt17kAMRXf2apRZFnY9GdRc_DmQsnMqqppGSkeIhMBRfTDHqz6sypJdWYEThJvXv18VUVUoRk1ST16dWXVSOnYiEkkWKS-vaq-2viVKEVQYj_yrYfjd6tzw">>},{<<"time">>,<<"2018-07-28T00:18:44.472Z">>},{<<"src">>,{[{<<"file">>,<<"/usr/lib/node_modules/spiegel/src/update-listeners.js">>},{<<"line">>,206},{<<"func">>,<<"_saveLastSeq">>}]}},{<<"v">>,0}]}

[error] 2018-07-28T00:18:44.536052Z couchdb@172.31.16.234 <0.218.0> -------- Daemon "spiegelChanges" made invalid request: {[{<<"name">>,<<"spiegel">>},{<<"hostname">>,<<"localhost">>},{<<"pid">>,10800},{<<"level">>,30},{<<"msg">>,<<"Requesting {\"url\":\"http://admin:**********@localhost:8080/test\",\"method\":\"PUT\",\"json\":{\"test\":\"test body\",\"change\":{\"_id\":\"01123abc33223241133dc5678a004701\",\"_rev\":\"1-967a00dff5e02add41819138abb3284d\"},\"db_name\":\"d\"}}">>},{<<"time">>,<<"2018-07-28T00:18:44.532Z">>},{<<"src">>,{[{<<"file">>,<<"/usr/lib/node_modules/spiegel/src/change-processor.js">>},{<<"line">>,107},{<<"func">>,<<"_request">>}]}},{<<"v">>,0}]}

[error] 2018-07-28T00:22:49.471558Z couchdb@172.31.16.234 <0.218.0> -------- Daemon "spiegelReplicator" made invalid request: {[{<<"name">>,<<"spiegel">>},{<<"hostname">>,<<"localhost">>},{<<"pid">>,10801},{<<"level">>,30},{<<"msg">>,<<"Beginning replication from http://admin:**********@localhost:5984/a to http://admin:**********@localhost:5984/b">>},{<<"time">>,<<"2018-07-28T00:22:49.470Z">>},{<<"src">>,{[{<<"file">>,<<"/usr/lib/node_modules/spiegel/src/replicators.js">>},{<<"line">>,195},{<<"func">>,<<"_replicate">>}]}},{<<"v">>,0}]
```
My first thought was the credentials used to make the requests, but I've double-checked them and they are correct.

I haven't been able to figure out what is causing this yet. I was testing things out on an AWS EC2 instance with a centos7 image and running couch 2.1.2. 
I have another instance that is setup the same way except it is running the previous version of centos7 (The centos7 image I'm using on AWS was recently updated and that previous version is no longer available) and couch 2.1.1. I haven't seen any of these issues on that machine. 
I tried setting up another machine running couch 2.1.1, and when I ran:
```
spiegel --type='install' --url='http://user:password@yourcouchdb.com:5984'    // obviously subbing in my info
```
it would create the spiegel DB but would just hang while it tried to create the _design docs

Wondering if you might have run into any issues like this before.

**Comment by redgeoff** [comment]
Hi @ephopper! I'm a little confused as to where these logs are coming from. They look like they may be CouchDB log entries, but if that is the case then they shouldn't include line numbers in the Spiegel code unless Spiegel is sending the error to CouchDB, something it should not be doing. Or, if these entries are from Spiegel then where is it getting names like _spiegelUpdates_?

I've personally tested Spiegel thoroughly with CouchDB 2.1.1 via [CouchDB running on Docker](https://github.com/redgeoff/redgeoff-couchdb-docker) and even run this setup in production and don't believe I have seen these errors before. I've been meaning to upgrade to CouchDB 2.1.2 so I'll try to find some time and give that a shot.

**Comment by ephopper** [comment]
Hi @redgeoff! Sorry for the confusion. I forgot to mention that I have couch running the Spiegel processes as OS daemons, so they are couch logs. spiegelUpdates, etc are just the names that I gave the processes. I had it set up that way so that couch could manage the processes.

I'm guessing there is a dependency issue somewhere along the line that I haven't been able to track down yet. In my original test environment, which is running the Spiegel processes as OS daemons with couch 2.1.1 on a previous release of Centos 7, I never saw these issues. Now that both couch and the Centos 7 image have been updated, these logs started showing up. 

From the tests I've done since opening this issue though, it looks like it might be a couch issue. If I run the processes outside of couch, the log output from Spiegel looks normal. 

I'll go ahead and close this issue, and see what I can figure out about couch 2.1.2.

Thanks!
